python/dgl/convert.py

```diff
@@ -158,16 +176,15 @@ def graph(data,
                        " Please refer to their API documents for more details.".format(
                            deprecated_kwargs.keys()))
 
-    u, v, urange, vrange = utils.graphdata2tensors(data, idtype)
+    (sparse_fmt, arrays), urange, vrange = utils.graphdata2tensors(data, idtype)
     if num_nodes is not None:  # override the number of nodes
         if num_nodes < max(urange, vrange):
             raise DGLError('The num_nodes argument must be larger than the max ID in the data,'
                            ' but got {} and {}.'.format(num_nodes, max(urange, vrange) - 1))
         urange, vrange = num_nodes, num_nodes
```

Comment:
Throw error if the user provided num_nodes does not match the inferred one for CSR input.

Reply:
Added the check in C.